src/App.js

```diff
@@ -103,6 +113,20 @@ class App extends Component {
       })
     }
 
+    filterByContinent = (event) => {
+      event.preventDefault();
+      let chosenContinent = event.target.name;
+      let countriesArr = this.state.continentData.continents.find(continent => {
+       if (continent.name === chosenContinent) {
+        return continent.countries;
+        }
+      });
+      let countriesMatchingFilter = countriesArr.countries;
+      this.setState({
+        countriesMatchingFilter: countriesMatchingFilter
+      })
+    }
+
```

Comment:
Great implementation  of this method, we almost gave up on this feature, but I am glad we are working towards getting this working.